### PR TITLE
hifiasm: ONT ultra-long integration

### PR DESCRIPTION
**Pull request by abueg**
hello! 👋🏼 

I started working on editing the wrapper for ONT UL integration, basing it off of the HiC integration. I think it's mostly working (including a test I'm using the `nanopore.fasta` from the flye test data for) but I have a problem with trying to use multiple ONT inputs... 

image

it seems to think the name of one file is `'input_0.fasta.gz input_1.fasta.gz'` instead of two files: `input_0.fasta.gz` and `input_1.fasta.gz`. 

if there isn't an easy workaround for this, i guess we could just have the user concatenate their inputs?

### CODE REVIEW

**Comment by bgruening** [comment]
Input staging problem: Test input file (nanopore.fasta.gz) cannot be found.

**Comment by bgruening** [comment]
grafik

Test 12 was killed, I guess it used to much memory?

**Comment by abueg** [comment]
trying with a subset of the nanopore input instead...

**Comment by abueg** [comment]
OK, disabling the initial bloom filter worked for test 12 to no longer be killed 👍🏼 ! 

I am wondering if you have any suggestions for how to account for multiple read files as input, though? (see error in first post)

File: tools/hifiasm/hifiasm.xml [hunk 1]
```diff
@@ -125,6 +134,16 @@
             --l-msjoin $hic_partition.l_msjoin
         #end if
 
+        #if str($ont_integration.ont_integration_selector) == 'set':
+            --ul '${ ' '.join(["%s" % $x for $x in $ultralong_inputs]) }'
```
Comment (bgruening) [inline tools/hifiasm/hifiasm.xml:138]: For your other question, I think this here is wrong - at least in combination with the outer `'`.

So what is happening here. the join is creating a string like `foo bar baz` and then you are putting `'` around it - yielding to `'foo bar baz'`. But you want `'foo' 'bar' 'baz'`.

Try something like that ... note the double inner quotes and the removed outer quotes.

`  ${ ' '.join(["'%s'" % $x for $x in $ultralong_inputs]) }  `

Comment (abueg) [reply tools/hifiasm/hifiasm.xml:138]: image
looks good! should i change the syntax on the HiC partition section to match as well?

Comment (bgruening) [reply tools/hifiasm/hifiasm.xml:138]: Yes please!

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
:heavy_check_mark:  thanks a lot!

**Comment by abueg** [comment]
of course, thank you for your help + patience!! 🙏🏼

**Pull request merged**